File: Bai_4/Bai4.c
//Name
//MSSV
//Class
//4.Liệt kê các số có số lần xuất hiện nhiều nhất trong mảng
#include<stdio.h>
#include<stdlib.h>

int n;
double *Tao_mang(){
    printf("Nhập kích thước mảng: ");
    while(scanf("%d", &n) != 1 || n < 0){
        printf("Nhập lại n: ");
        while(getchar() != '\n');
    }
    double *arr = malloc(n * sizeof(double));
    if (arr == NULL) return NULL;
    for(int i = 0; i < n; i++){
        printf("Giá trị của arr[%d] = ", i);
        while(!scanf("%lf", &arr[i])){
            printf("Nhập lại: ");
            while(getchar() != "\n");
        }    
    }
    return arr;
}
int main(){
    double *arr = Tao_mang();

}

File: Bai_4/Bai4.py
#Name
#MSSV
#Class
#4.Liệt kê các số có số lần xuất hiện nhiều nhất trong mảng

n = int(input('Nhập kích thước mảng: '))
arr = []
for i in range(n):
    arr.append(int(input(f"Giá trị của arr[{i}] là: ")))
print("\n",arr)
table = dict()
dem = 1
for i in arr:
    if i not in table:
        table[i] = 1
    else:
        table[i] += 1
        if(table[i] > dem):
            dem = table[i]
print(table)
for u,v in table.items():
    if v == dem:
        print(f"{u} xuất hiện nhiều nhất với số lần là {v}")
        
